packages/vats/src/orchestration.js

```diff
@@ -0,0 +1,268 @@
+          // @ts-expect-error Fail does not satisfy possibly undefined
+          return E(connection).close();
+        },
+      },
+      connectionHandler: {
+        /**
+         * @param {Connection} connection
+         * @param {string} localAddr
+         * @param {string} remoteAddr
+         * @param {ConnectionHandler} _connectionHandler
+         */
+        async onOpen(connection, localAddr, remoteAddr, _connectionHandler) {
+          trace(`ICA Channel Opened for ${localAddr} at ${remoteAddr}`);
+          this.state.connection = connection;
+          this.state.remoteAddress = remoteAddr;
+          this.state.localAddress = localAddr;
+          // XXX parseAddress currently throws, should it return '' instead?
+          this.state.accountAddress = parseAddress(remoteAddr);
+        },
+        async onClose(_connection, reason) {
+          trace(`ICA Channel closed. Reason: ${reason}`);
+          // XXX handle connection closing
+        },
+        async onReceive(connection, bytes) {
+          trace(`ICA Channel onReceive`, connection, bytes);
+          return '';
+        },
+      },
+    },
+  );
+
+export const OrchestrationI = M.interface('Orchestration', {
+  createAccount: M.callWhen(M.string(), M.string())
+    .optional(M.remotable('Port'))
+    .returns(M.remotable('ChainAccount')),
+  getChain: M.callWhen(M.string()).returns(M.remotable('Chain')),
+});
+
+/**
+ * @param {import('@agoric/base-zone').Zone} zone
+ * @param {ReturnType<typeof prepareChainAccount>} createChainAccount
+ */
+const prepareOrchestration = (zone, createChainAccount) =>
+  zone.exoClassKit(
+    'Orchestration',
+    {
+      admin: M.interface('OrchestrationAdmin', {
+        placeholder: M.call().returns(M.string()),
+      }),
+      public: OrchestrationI,
+    },
+    /** @param {Partial<OrchestrationPowers>} [initialPowers] */
+    initialPowers => {
+      /** @type {PowerStore} */
+      const powers = zone.detached().mapStore('PowerStore');
+      if (initialPowers) {
+        for (const [name, power] of Object.entries(initialPowers)) {
+          powers.init(/** @type {keyof OrchestrationPowers} */ (name), power);
+        }
+      }
+      return { powers, icaControllerNonce: 0 };
+    },
+    {
+      admin: {
+        placeholder() {
+          return 'test';
+        },
+      },
+      public: {
+        /**
+         * @param {import('@agoric/orchestration').ConnectionId} hostConnectionId
+         *   the counterparty connection_id
+         * @param {import('@agoric/orchestration').ConnectionId} controllerConnectionId
+         *   self connection_id
+         * @param {Port} [currentPort] if a port is provided, it will be reused
+         *   to create the account
+         * @returns {Promise<ChainAccount>}
+         */
+        async createAccount(
+          hostConnectionId,
+          controllerConnectionId,
+          currentPort,
+        ) {
+          await null;
+
+          let port;
```

Comment:
`port` never changes and you're figuring out what to initialize it with. let's do the work to make it a `const`.